aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AWSPinpointAnalyticsPluginBehavior.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.analytics.pinpoint
+
+import com.amplifyframework.analytics.AnalyticsBooleanProperty
+import com.amplifyframework.analytics.AnalyticsCategoryBehavior
+import com.amplifyframework.analytics.AnalyticsDoubleProperty
+import com.amplifyframework.analytics.AnalyticsEventBehavior
+import com.amplifyframework.analytics.AnalyticsIntegerProperty
+import com.amplifyframework.analytics.AnalyticsProperties
+import com.amplifyframework.analytics.AnalyticsStringProperty
+import com.amplifyframework.analytics.UserProfile
+import java.lang.IllegalArgumentException
+
+internal class AWSPinpointAnalyticsPluginBehavior(
+    val analyticsClient: AnalyticsClient
+) : AnalyticsCategoryBehavior {
+
+    override fun identifyUser(userId: String, profile: UserProfile?) {
+        TODO("Not yet implemented")
+    }
+
+    override fun disable() {
+        TODO("Not yet implemented")
+    }
+
+    override fun enable() {
+        TODO("Not yet implemented")
+    }
+
+    override fun recordEvent(eventName: String) {
+        val pinpointEvent = analyticsClient.createEvent(eventName)
+        analyticsClient.recordEvent(pinpointEvent)
+    }
+
+    override fun recordEvent(analyticsEvent: AnalyticsEventBehavior) {
+        val attributes = mutableMapOf<String, String>()
+        val metrics = mutableMapOf<String, Double>()
+        analyticsEvent.properties.forEach { property ->
+            val key = property.key
+            val analyticsProperty = property.value
+            when (analyticsProperty.value) {
+                is AnalyticsStringProperty -> {
```

Comment:
🤯

Reply:
Was going to ask, do you have to re-cast on L56. Typically, when using `is` in a when statement, you don't have to cast again.